### PR TITLE
Add theme

### PR DESCRIPTION
**Pull request by tushar-rishav**
Option to toggle between two Light and Dark theme

[Demo](https://cloud.githubusercontent.com/assets/7397433/15978692/7a540380-2f7d-11e6-8510-1f13b68d9a1a.gif)

### CODE REVIEW

File: coalahtml/_coalahtml/app/controllers/files.js [hunk 1]
```diff
@@ -3,6 +3,26 @@
...
+      for (var index = rootHomeLen; index < fileNameLen; index++) {
+        link = link + '/' + files[index];
+        linkNameContainer.push({
+          'name' : files[index] + (index < fileNameLen-1 ? '/' : ''),
+          'link' : link
+        });
+      }
+      $scope.linkNameContainer = linkNameContainer;
+    };
     var parseCoalaProject = function(fileName) {
```
Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/controllers/files.js]: :point_up:  These changes are bug fix for last PR. Hence added with this PR.

File: coalahtml/_coalahtml/app/views/files.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <div class="files">
   <div class="dir-contents" ng-if="fileType === 'dir'">
-    <h3>
+    <h3 ng-class="{'dark':!THEME}">
       Contents of
-      <span ng-repeat= "file in getNameLink(fileName)" class="filename">
+      <span ng-repeat= "file in linkNameContainer" class="filename">
```
Comment (tushar-rishav) [inline coalahtml/_coalahtml/app/views/files.html]: This is the fix for reaching maximum iteration limit.

**Comment by sils1297** [comment]
coverage low, btw. you seem to be lacking some coverage anyway

**Comment by Adrianzatreanu** [comment]
seems travis times out ;/

**Comment by Adrianzatreanu** [comment]
marking wip as lacking tons of coverage and travis timing out.

File: coalahtml/_coalahtml/app/views/files.html (hunk 2 above)
Comment (Uran198) [inline coalahtml/_coalahtml/app/views/files.html]: This fix should be in separate commit.

File: tests/specs/controllers/FilesCtrlSpec.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        it('test fileBack is obtained correctly', function() {
+            expect(scope.fileBack).to.equal('/home');
+        });
+        it('test fileType', function() {
+            expect(scope.fileType).to.equal('dir');
+        });
+        it('test fileContents', function() {
+            expect(scope.fileContents[0].name).to.equal('/home/Desktop/test.py');
+            expect(scope.fileContents[0].type).to.equal('file');
+            expect(scope.fileContents[0].result).to.true;
```
Comment (Uran198) [inline tests/specs/controllers/FilesCtrlSpec.js]: It's better to compare the whole list, that you'll make sure that there is nothing redundant as well, something like:
```
expect(scope.fileContents).to.equal(...);
```

File: tests/specs/controllers/FilesCtrlSpec.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        beforeEach(inject(function ($controller) {
+            $controller('FilesCtrl', {
+                $scope: scope,
+                $routeParams: routeParams,
+                $rootScope: rootScope
+            });
+        }));
+        it('test fileName link', function() {
+            expect(scope.linkNameContainer).to.have.length(2);
+            expect(scope.linkNameContainer[1].name).to.equal('Desktop');
```
Comment (Uran198) [inline tests/specs/controllers/FilesCtrlSpec.js]: It's better to test the whole list, not only parts of it.

File: tests/specs/controllers/MainCtrlSpec.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+          $httpBackend.verifyNoOutstandingExpectation();
+          $httpBackend.verifyNoOutstandingRequest();
+        });
+
+        it('test result count', function() {
+            $httpBackend.flush();
+            expect(rootScope.resultsCount).to.equal(1);
+        });
+        it('test parseResult', function() {
+            $httpBackend.flush();
```
Comment (Uran198) [inline tests/specs/controllers/MainCtrlSpec.js]: You can move this `flush` to the end of `beforeEach` as it's used in every test case.

File: tests/specs/controllers/MainCtrlSpec.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            expect(rootScope.resultFiles).to.have.property("/home/Desktop/test.py")
+              .with.length(1);
+            expect(rootScope.resultFiles["/home/Desktop/test.py"][0].start).to.equal(33);
```
Comment (Uran198) [inline tests/specs/controllers/MainCtrlSpec.js]: It's better to test the whole objects, not only parts of them.

File: tests/specs/controllers/MainCtrlSpec.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        });
+        it('test scope.data', function() {
+            $httpBackend.flush();
+            expect(scope.data).to.have.length(1);
+            expect(scope.data[0]).to.equal(rootScope.resultFiles);
```
Comment (Uran198) [inline tests/specs/controllers/MainCtrlSpec.js]: It's better to test the whole objects, not only parts of them.

File: tests/specs/controllers/MainCtrlSpec.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+            });
+        }));
+        afterEach(function() {
+          $httpBackend.verifyNoOutstandingExpectation();
+          $httpBackend.verifyNoOutstandingRequest();
+        });
+
+        it('test result count', function() {
+            $httpBackend.flush();
+            expect(rootScope.resultsCount).to.equal(1);
```
Comment (Uran198) [inline tests/specs/controllers/MainCtrlSpec.js]: It's better to test the whole objects, not only parts of them.

Comment (tushar-rishav) [inline tests/specs/controllers/MainCtrlSpec.js]: Oh didn't see all these comments. Will address them in next push. :+1:

File: tests/specs/controllers/MainCtrlSpec.js (hunk 7 above)
Comment (tushar-rishav) [inline tests/specs/controllers/MainCtrlSpec.js]: Um here data has only one object i.e resultFiles. So I guess maybe it's okay here?

File: tests/specs/controllers/FilesCtrlSpec.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+            expect(scope.fileName).to.equal(routeParams.fileName);
+        });
+        it('test fileBack is obtained correctly', function() {
+            expect(scope.fileBack).to.equal('/home');
+        });
+        it('test fileType', function() {
+            expect(scope.fileType).to.equal('dir');
+        });
+        it('test fileContents', function() {
+            expect(scope.fileContents[0]).to.deep.equal(FILE_CONTENTS);
```
Comment (Uran198) [inline tests/specs/controllers/FilesCtrlSpec.js]: It may be better to compare the whole list.

Comment (Uran198) [inline tests/specs/controllers/FilesCtrlSpec.js]: It also may be better to compare the whole scope.

File: tests/specs/controllers/MainCtrlSpec.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+        it('test result count', function() {
+            expect(rootScope.resultsCount).to.equal(1);
+        });
+        it('test parseResult', function() {
+            expect(rootScope.resultFiles).to.deep.equal(RESULT_FILES);
+        });
+        it('test scope.data', function() {
+            expect(scope.data).to.have.length(1);
+            expect(scope.data[0]).to.deep.equal(rootScope.resultFiles);
```
Comment (Uran198) [inline tests/specs/controllers/MainCtrlSpec.js]: can it be like `expect(scope.data).to.deep.equal([rootScope.resultFiles]);`?

**Comment by Uran198** [comment]
ack d11966e2329aa785b9da1c4690d75fe487754521 a5a28f9aa5a2563215578913781b1d94aa477d00 fb89ab79d6a2b49666ae2496b2dd6042ce5e02fb

File: tests/specs/controllers/MainCtrlSpec.js [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        });
+
+        it('test result count', function() {
+            expect(rootScope.resultsCount).to.equal(1);
+        });
+        it('test parseResult', function() {
+            expect(rootScope.resultFiles).to.deep.equal(RESULT_FILES);
+        });
+        it('test scope.data', function() {
+            expect(scope.data).to.have.length(1);
```
Comment (Uran198) [inline tests/specs/controllers/MainCtrlSpec.js]: This line is redundant.

**Comment by Uran198** [comment]
ack 80288555ab67e4871c5fa468455b240548c02229 25ea0ef72a19936b4758a0b6c4ae1e079ba6ea91 6fbae28f4067aef699e072353f008d8964407cc5 e4de2672b697ab73da4784b5b1bd3149683fac7b

**Comment by tushar-rishav** [comment]
reack 95b205c

**Comment by tushar-rishav** [comment]
@rultor merge
